node/external/transactionAPI/refundDetector.go

```diff
@@ -0,0 +1,45 @@
+package transactionAPI
+
+import (
+	"bytes"
+	"encoding/hex"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type refundDetectorInput struct {
+	Value         string
+	Data          []byte
+	ReturnMessage string
+	GasLimit      uint64
+}
+
+type refundDetector struct {
+}
+
+func newRefundDetector() *refundDetector {
+	return &refundDetector{}
+}
+
+// Also see: https://github.com/ElrondNetwork/elastic-indexer-go/blob/master/process/transactions/scrsDataToTransactions.go
```

Comment:
wondering how we can remove all these code duplicates

Reply:
One option would be to create a new repository (and go package), such as `transaction-decoder` or `transaction-classifier`, independent of any other package. Then, we would reference this package in `elrond-go` and `elastic-indexer-go`.

The newly created repository / package would be quite stable in practice.